src/speakers/extraction.py

```diff
@@ -329,3 +332,339 @@ def get_speakers_in_bold(doc):
+    conversation=[]
+    for line in lines:
+        if line not in names:
+            s=s+line+' '
+        else:
+            conversation.append(s)
+            s=''
+    if s!='':
+        conversation.append(s)
+    return (conversation)
+
+
+def modify_str(l):
+    str=l.replace('.','. ')
+    l=l.replace(',',', ')
+    l=l.replace('from','from ')
+    l=l.replace('(','')
+    l=l.replace(')','.')
+    w=l.split()
+    return w
+
+#Function tries to find that paragraph where the full name occurs
+def find_count(name,w,count_max):
+    flag=0
+    position=0
+    count=0
+    name_index=0
+    for word in name.split():
+        word=remove_last_char_if_colon(word)
+        if word in w:
+            if flag==0:
+                name_index=w.index(word)
+                flag=1
+            count=count+1
+        elif flag==0:
+            position=position+1
+    if count>count_max:
+        count_max=count
+    return (count,count_max,position,name_index)
+
+
+def check_if_firm(firm):
+    count=0
+    #if we Identify a firm that has more than 4 words and all of the words belong to a English Dictionary, we discard it from being a firm
+    if len(firm.split())>4:
+        for word in firm.split():
+            if english_dictionary.check(word):
+                count=count+1
+        if count==len(firm.split()):
+            firm=''
+    if firm=='Thank you':
+        firm=''
+    return firm
+
+def pass_1(names,conversation,order_of_speakers):
+    final={}
+    #first value in conversation contains lines before any speaker starts, hence we filter it out
+    conversation=conversation[1:len(conversation)]
+    words_not_found_as_first_word_of_firm=['Sir,','I','Mr.']
+    for name in names:
+        name=remove_last_char_if_colon(name)
+        #count_max stores the maximum number of words from a name that are Identified in any paragraph
+        count_max=0
+        for i in range(len(conversation)):
+            if order_of_speakers[i]=='Moderator:' or order_of_speakers[i]=='Moderator' or order_of_speakers[i]=='Operator' or order_of_speakers[i]=='Operator:':
+                count=0
+                l=conversation[i]
+                #modify_str takes the conversation, makes some changes and returns the words to w
+                w=modify_str(l)
+                #postion is the postion of the word identified in name 
+                #name_index is the postion of the word identified in w                
+                count,count_max,position,name_index=find_count(name,w,count_max)
+                #if more than 50% of words in a name are Identified, and words Identified are less than max Identified till now, we search for the firm
+                if count/len(name.split())>=0.5 and count>=count_max:
+                    #flag_2 is used to check if We've found any work Starting with a Capital
+                    #flag_3 is used to check if We've reached end of Conversation while adding names to the firm
+                    flag_2=0
```

Comment:
nit: We can use more meaningful names instead of flag.

Reply:
Meaning is explained in the comment and variables not used that often